### DISCUSSION TITLE
Use sleep when raise exception asyncio.TimeoutError, KafkaError

### DISCUSSION THREAD

**Discussion by mashkovd** [aio-libs/aiokafka#949] [General]
Hello,

I've identified an issue related to Kafka unavailability, where the library logs numerous messages with the text "Unable to connect to node with id %s: %s." To address this, I'm considering creating a pull request. However, I want to ensure that I'm not overlooking any crucial details.

In [this](https://github.com/aio-libs/aiokafka/blob/d7201c1e26fd2f3e51f15a2cdeda9753d4fd3a1a/aiokafka/consumer/fetcher.py#L663) section, the code uses `await asyncio.sleep(self._retry_backoff)` to introduce a delay, and the retry interval is controlled by the retry_backoff_ms parameter.

I propose incorporating a similar approach in [this](https://github.com/aio-libs/aiokafka/blob/d7201c1e26fd2f3e51f15a2cdeda9753d4fd3a1a/aiokafka/client.py#L460) section. Does this seem like the right solution to you?

**Comment by jaredvacanti** [comment]
Hi @mashkovd, I'm also running into this error. I tried putting the `await asyncio.sleep(self._retry_backoff)` you recommended at https://github.com/aio-libs/aiokafka/blob/master/aiokafka/client.py#L474 and it seems to solve the logging loop I was experiencing.

Have you had any negative experiences otherwise? Would you consider submitting a pull with the fix? It does work for me.

**Reply by jaredvacanti** [reply]
Looks like https://github.com/aio-libs/aiokafka/pull/1014 will fix this.
